packages/mui-base/src/Checkbox/CheckboxIndicator.test.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import { createRenderer } from '@mui/internal-test-utils';
+import { describeConformance } from '../../test/describeConformance';
+import { Checkbox } from '.';
+import { CheckboxContext } from './CheckboxContext';
+
+const testContext = {
+  checked: true,
+  disabled: false,
+  readOnly: false,
+  required: false,
+  indeterminate: false,
+};
+
+describe('<Checkbox.Indicator />', () => {
+  const { render } = createRenderer();
+
+  describeConformance(<Checkbox.Indicator />, () => ({
+    inheritComponent: 'span',
+    refInstanceof: window.HTMLSpanElement,
+    render(node) {
+      return render(
+        <CheckboxContext.Provider value={testContext}>{node}</CheckboxContext.Provider>,
+      );
+    },
+    skip: ['reactTestRenderer'],
+  }));
+
+  it('should not render indicator by default', () => {
```

Comment:
I wonder if it's worth having an indicator present at all times - to render a different content when the checkbox is unchecked.

Reply:
By default, it's useful to have it conditionally unmount so it doesn't render the children without needing to hide it with CSS. `keepMounted` lets you implement CSS animations